test/extended/registry/registry.go

```diff
@@ -0,0 +1,145 @@
+package registry
+
+import (
+	"fmt"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	dockerClient "github.com/fsouza/go-dockerclient"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/util/wait"
+
+	regclient "github.com/openshift/origin/pkg/dockerregistry"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+	imagesutil "github.com/openshift/origin/test/extended/images"
+	exutil "github.com/openshift/origin/test/extended/util"
+	testutil "github.com/openshift/origin/test/util"
+)
+
+const (
+	repoName  = "app"
+	tagName   = "latest"
+	imageSize = 1024
+)
+
+var _ = g.Describe("[registry][migration] manifest migration from etcd to registry storage", func() {
+	defer g.GinkgoRecover()
+	var oc = exutil.NewCLI("registry-migration", exutil.KubeConfigPath())
+
+	// needs to be run at the top of each It; cannot be run in AfterEach which is run after the project
+	// is destroyed
+	tearDown := func(oc *exutil.CLI) {
+		deleteTestImages(oc)
+	}
+
+	g.It("registry can get access to manifest", func() {
```

Comment:
Not at all - I was able to clearly read through it and understand exactly what it was testing and the scope.  More extended tests should be this easy to read.